core/models/asset.py

```diff
@@ -286,3 +286,17 @@ def save_current_asset_owner(sender, **kwargs):
     asset_owner = kwargs.get('instance')
     asset_owner.asset.assigned_to = asset_owner.current_owner
     asset_owner.asset.save()
+
+
+@receiver(post_save, sender=AllocationHistory)
+def update_asset_status(sender, **kwargs):
+    allocation_history = kwargs.get('instance')
+    asset = allocation_history.asset
+    if asset.assigned_to and asset.current_status == AVAILABLE:
+        asset_status = AssetStatus.objects.get(
```

Comment:
Here we should create a new `AssetStatus` object. This will take care of the asset's current status